src/vs/vscode.proposed.d.ts

```diff
@@ -167,4 +167,20 @@ declare module 'vscode' {
 		 */
 		export function deleteSecret(service: string, account: string): Thenable<boolean>;
 	}
+
+	export namespace window {
+
+		/**
+		 * Whether the current window is focused.
+		 *
+		 * @readonly
+		 */
+		export let isFocused: boolean;
```

Comment:
Alternative idea would be have an enum ala `WindowState` with `focused`, `visible`, 'hidden` etc. That will be more future proof because less verbose when folks ask for those things.